src/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,210 @@
+import { generateRandomId } from "@utilities/generateRandomId";
+import { useOverlayTriggerState } from "@react-stately/overlays";
+import { PatternDesign, PatternTheme } from "@foundation/Pattern";
+import { IconAcademy, IconAudio, IconIcons } from "@foundation/Icon";
+import { ScrollWrapperDirection } from "@components/ScrollWrapper/types";
+import { OverlayContainer, OverlayProvider } from "@react-aria/overlays";
+import { ModalHeaderProps, ModalHeaderVariant, ModalProps, ModalVisualProps } from "./types";
+import { FormControl, FormControlDirection, FormControlStyle } from "@components/FormControl";
+
+// eslint-disable-next-line import/no-default-export
+export default {
+    title: "Components/Modal",
+    component: Modal,
+    argTypes: {
+        pattern: {
+            table: {
+                category: "Visual",
+            },
+            name: "Pattern",
+            options: ["None", ...Object.values(PatternDesign)],
+            mapping: { None: null, ...Object.values(PatternDesign) },
+            defaultValue: PatternDesign.DigitalAssets,
+            control: { type: "select" },
+        },
+        foregroundColor: {
+            table: {
+                category: "Visual",
+            },
+            name: "Theme",
+            options: ["None", ...Object.values(PatternTheme)],
+            mapping: { None: null, ...Object.values(PatternDesign) },
+            defaultValue: PatternTheme.Black,
+            control: { type: "select" },
+        },
+        title: {
+            table: {
+                category: "Header",
+            },
+            name: "Title",
+            defaultValue: "Awesome modal",
+            control: { type: "text" },
+        },
+        leadText: {
+            table: {
+                category: "Header",
+            },
+            name: "Lead Text",
+            defaultValue: "The Home for Brand Management",
+            control: { type: "text" },
+        },
+        decorator: {
+            table: {
+                category: "Header",
+            },
+            name: "Decorator",
+            options: ["None", "Icons", "Academy", "Audio"],
+            mapping: {
+                None: null,
+                Icons: <IconIcons key="iconIcons" />,
+                Academy: <IconAcademy key="iconAcademy" />,
+                Audio: <IconAudio key="iconAudio" />,
+            },
+            defaultValue: "Icons",
+            control: { type: "select" },
+        },
+        variant: {
+            table: {
+                category: "Header",
+            },
+            name: "Variant",
+            options: ["None", ...Object.values(ModalHeaderVariant)],
+            mapping: { None: null, ...Object.values(ModalHeaderVariant) },
+            defaultValue: ModalHeaderVariant.Default,
+            control: { type: "select" },
+        },
+    },
+    decorators: [
+        (Story) => (
+            <OverlayProvider>
+                <OverlayContainer>
```

Comment:
Would be cool to not have to wrap our whole application in those Provider and Container

Reply:
Hey @SamuelAlev 

@jonasbuechel asked me to have a look at the PR. From what I understand, unfortunately we won't get around that.

> **The application is contained in an OverlayProvider**, which is used to hide the content from screen readers with aria-hidden while a modal is open. In addition, each modal **must be contained in an OverlayContainer**, which uses a React Portal to render the modal at the end of the document body.
>
> Note: `useModal` only hides content within parent OverlayProvider components. [...]

https://react-spectrum.adobe.com/react-aria/useDialog.html

Reply:
We can not wrap our app where we use Terrific with some React components, it would make `React -> Terrific -> React`.
In full React page (marketplace, brand listening, ...) it should be ok however.

Reply:
If that's an issue I only see three ways going forward:

- Find a library that does it differently and redo this PR
- Implement a solution without any third-party dependencies
- Find a way for the Spectrum components to work the way you need in Clarify

I'll approve the PR from my side, but it's probably your decision on how you want to go forward with it. I'm also unsure about the importance of a11y in Clarify during the Terrific → React transition phase.

Reply:
We can use the modal component without OverlayProvider as well, but we will lose the accessibility optimizations this way. I'll give it a try within Terrific. As the whole Terrific app is in bad shape accessibility-wise, I think we can live with this compromise for legacy views.

Reply:
It works within a react component in clarify 👍 
<img width="1331" alt="Screenshot 2022-03-31 at 11 16 06" src="https://user-images.githubusercontent.com/3276065/161021334-8436a5a8-d782-4e47-8d02-8cd267428856.png">